node-cli/users.js

```diff
@@ -0,0 +1,106 @@
+const inquirer = require("inquirer");
+const Realm = require("realm");
+const index = require("./index");
+const config = require("./config");
+const main = require("./main");
+const output = require("./output");
+
+const REALM_APP_ID = config.realmAppId;
+const appConfig = {
+  id: REALM_APP_ID,
+  timeout: 10000,
+};
+
+const app = new Realm.App(appConfig);
+
+authedUser = {};
+
+async function logIn() {
+  const input = await inquirer.prompt([
+    {
+      type: "input",
+      name: "email",
+      message: "Email:",
+    },
+    {
+      type: "password",
+      name: "password",
+      message: "Password:",
+      mask: "*",
+    },
+  ]);
+
+  try {
+    const credentials = Realm.Credentials.emailPassword(
+      input.email,
+      input.password
+    );
+    const user = await app.logIn(credentials);
+    if (user) {
+      authedUser = user;
+      output.result("You have successfully logged in as " + authedUser.id);
+      return main.mainMenu();
+    } else {
+      output.error("There was an error logging you in");
+      return logIn();
+    }
+  } catch (err) {
+    output.error(JSON.stringify(err, " ", 3));
+    return logIn();
+  }
+}
+
+async function registerUser() {
+  output.header("WELCOME, NEW USER");
+  const input = await inquirer.prompt([
+    {
+      type: "input",
+      name: "email",
+      message: "Email:",
+    },
+    {
+      type: "password",
+      name: "password",
+      message: "Password:",
+      mask: "*",
+    },
+  ]);
+
+  try {
+    const result = await app.emailPasswordAuth.registerUser(
+      input.email,
+      input.password
+    );
+    output.result(result);
+    const user = await logIn(input.email, input.password);
```

Comment:
* This should be `app.logIn()` (Realm.App method) not `logIn()` (the CLI view)
* This should accept a Credential rather than the email/password as individual args
* If `app.logIn()` resolves then it was successful, otherwise it throws. We should handle login errors in the catch block.


```js
  try {
    const { email, password } = input;
    await app.emailPasswordAuth.registerUser(email, password);
    const credentials = Realm.Credentials.emailPassword(email, password);
    await app.logIn(credentials);
    output.result(
      "You have successfully created a new Realm user and are now logged in."
    );
    return main.mainMenu();
  } catch (err) {
    output.error("There was an error registering the new user account.");
    output.error(JSON.stringify(err, " ", 3));
    return registerUser();
  }
```